### ISSUE TITLE
Server: errors relating to MD057 can persist in the editor even if fixed

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#517]
### rumdl version

0.1.45

### Installation method

mise

### Bug description

Rule MD057 (https://github.com/rvben/rumdl/blob/a304d288c836b694e792d329a782db3e3165858c/docs/md057.md) helps ensure that relative Markdown links are valid. This is very helpful in an editor (though `rumdl server`) because you can get live feedback as you type.

However, in some cases, even if a link is fixed, the editor isn't notified of it and the violation remains.

### Steps to reproduce

1. Create two Markdown file.
   - First Markdown file:
     ```markdown
     ---
     title: Heading
     ---
     
     Here is [a link](linked.md).
     ```
   - Second Markdown file (called `linked.md`) in the same directory as the first file:
     ```markdown
     ---
     title: Heading
     ---
     
     Here's some Markdown.
     ```
1. Open the first Markdown file in an editor that's configured for `rumdl server`.
1. Update the first Markdown file in the editor to be:
   ```markdown
   ---
   title: Heading
   ---
   
   Here is [a link](../linked.md).
   ```
   The editor will show an error that the link is invalid.
1. Restore the first Markdown file to its previous state. The error is still present in the editor.
1. Run `rumdl check` on the first Markdown file. No error is returned.

### Expected behavior

Given the above scenario, I'd expect the editor (through `rumdl server`) to return an error, then remove the error when the link is fixed.

### Actual behavior

Given the above scenario, the error persists. It appears that the editor (through the LSP client) isn't notified that the fix is in place.

### Configuration

None.

### Operating system

macOS

**Comment by rvben** [comment]
Thanks for reporting this! Fixed in v0.1.47. The server now publishes empty diagnostics to invalidate the client cache when switching to pull mode, preventing stale errors from persisting in the editor.

**Comment by eread** [comment]
> Thanks for reporting this! Fixed in v0.1.47. The server now publishes empty diagnostics to invalidate the client cache when switching to pull mode, preventing stale errors from persisting in the editor.

Thanks @rvben. I find this rock-solid if you invalidate the link by appending text, but not if you prepend text or add text in the middle. Here's a different scenario:

1. Given the scenario above,  modify the link to this: `[a link](linked.mda)`, then correct it. Seems to work great.
1. Given the scenario above,  modify the link to this: `[a link](linkeda.md)`, then correct it. Problem seems stuck.
1. Given the scenario above,  modify the link to this: `[a link](alinked.md)`, then correct it. Problem seems stuck.

**Comment by eread** [comment]
> > Thanks for reporting this! Fixed in v0.1.47. The server now publishes empty diagnostics to invalidate the client cache when switching to pull mode, preventing stale errors from persisting in the editor.
> 
> Thanks [@rvben](https://github.com/rvben). I find this rock-solid if you invalidate the link by appending text, but not if you prepend text or add text in the middle. Here's a different scenario:
> 
> 1. Given the scenario above,  modify the link to this: `[a link](linked.mda)`, then correct it. Seems to work great.
> 2. Given the scenario above,  modify the link to this: `[a link](linkeda.md)`, then correct it. Problem seems stuck.
> 3. Given the scenario above,  modify the link to this: `[a link](alinked.md)`, then correct it. Problem seems stuck.

@rvben Would you prefer if I raise a new issue, or are you happy to reopen this one? 🙇🏻

**Comment by eread** [comment]
> > > Thanks for reporting this! Fixed in v0.1.47. The server now publishes empty diagnostics to invalidate the client cache when switching to pull mode, preventing stale errors from persisting in the editor.
> > 
> > 
> > Thanks [@rvben](https://github.com/rvben). I find this rock-solid if you invalidate the link by appending text, but not if you prepend text or add text in the middle. Here's a different scenario:
> > 
> > 1. Given the scenario above,  modify the link to this: `[a link](linked.mda)`, then correct it. Seems to work great.
> > 2. Given the scenario above,  modify the link to this: `[a link](linkeda.md)`, then correct it. Problem seems stuck.
> > 3. Given the scenario above,  modify the link to this: `[a link](alinked.md)`, then correct it. Problem seems stuck.
> 
> [@rvben](https://github.com/rvben) Would you prefer if I raise a new issue, or are you happy to reopen this one? 🙇🏻

@rvben I've raised: https://github.com/rvben/rumdl/issues/548 because I've figured out a simpler way to explain it.
